.devcontainers/devcontainer.json

```diff
@@ -0,0 +1,41 @@
+// For format details, see https:https://containers.dev/implementors/spec/. For config options, see the README at:
+// https://github.com/devcontainers/images/tree/main/src/base-debian
+{
+    "name": "Kubernetes with e2e testing image build and push",
+    "image": "mcr.microsoft.com/devcontainers/base:bullseye",
+    // Setup the go environment and mount into the dev container at the expected location
+    "hostRequirements": {
+        "cpus": 4
+    },
+    // Copy over welcome message and install pyyaml
+    "onCreateCommand": "bash .devcontainer/setup.sh",
+    // for Kubernetes testing, suppress extraneous forwarding messages
+    "otherPortsAttributes": {
+        "onAutoForward": "silent"
+    },
+    "remoteUser": "root",
+    // Install utils, go, docker-in-docker, and other tools
+    // For more info about dev container Features see https://containers.dev/features
+    "features": {
+        "ghcr.io/devcontainers/features/go:1": {
+            "version": "1.21.1",
```

Comment:
Same here, does this need to be consistent with the go version we are using in the rest of the repo?

Reply:
If we use our own base image, is this required?

Reply:
We struggled with this. The problem being that this gets installed before the repo is downloaded. I could make a custom feature that pre-downloads the repo to check the right go version, but I thought that was a lot for an initial commit?

Reply:
Because `kube-cross` includes the correct go version, the go feature would not be needed. A win!